docs/topics/proc_exposing-application-metrics-using-prometheus-with-vertx.adoc

```diff
@@ -0,0 +1,190 @@
+
+[id='exposing-application-metrics-using-prometheus-with-vertx_{context}']
+= Exposing application metrics using Prometheus with {Vertx}
+
+In this example, you:
+
+* Configure your application to expose metrics.
+* Collect and view the data using Prometheus.
+
+Prometheus connects to a monitored application to collect data; the application does not send metrics to a server.
+
+.Prerequisites
+
+* Your application configured to use `vertx-maven-plugin`.
+For more information, see xref:configuring-your-application-to-use-vertx_{context}[].
+* Prometheus configured to collect metrics from the application:
++
+--
+. Download and extract the link:https://prometheus.io/download/[archive^] with the latest Prometheus release:
++
+[source,bash,options="nowrap"]
+----
+$ wget https://github.com/prometheus/prometheus/releases/download/v2.4.3/prometheus-2.4.3.linux-amd64.tar.gz
+$ tar -xvf  prometheus-2.4.3.linux-amd64.tar.gz
+----
+
+. Navigate to the directory with Prometheus:
++
+[source,bash,options="nowrap"]
+----
+$ cd  prometheus-2.4.3.linux-amd64
+----
+
+. Append the following snippet to the `prometheus.yml` file to make Prometheus automatically collect metrics from your application:
++
+[source,yaml,options="nowrap"]
+----
+  scrape_configs:
+    - job_name: 'prometheus'
+      static_configs:
+      - targets: ['localhost:9090']
+    - job_name: 'vertx-app'
+      static_configs:
+      - targets: ['localhost:8081']
+----
++
+The default behavior of {Vertx}-based applications is to expose metrics at the `/metrics` endpoint.
+This is what Prometheus expects.
+--
+
+* The Prometheus server started on `localhost`:
++
+--
+Start Prometheus and wait until the `Server is ready to receive web requests` message is displayed in the console.
+
+[source,bash,options="nowrap"]
+----
+$ ./prometheus
+----
+--
+
+
+.Procedure
+
+. Include the `vertx-micrometer`, `micrometer-registry-prometheus` and `vertx-web` dependencies in the `pom.xml` file of your application:
++
+.pom.xml
+[source,xml]
+----
+<dependency>
+  <groupId>io.vertx</groupId>
+  <artifactId>vertx-micrometer-metrics</artifactId>
+  <version>3.5.4</version>
```

Comment:
I don't know if we want to have Vert.x version hard-coded here. You'd need to update each time there's a release and it should come from `vertx-dependencies` anyway :)

Reply:
Oh, I missed that, thanks. Does `${vertx.version}` work?

Reply:
It should work by removing `<version></version>` tag altogether, as version is then automatically pulled/aligned from `vertx-dependencies` in project's dependency management section. How to set that up is described in 2. procedure step here https://launcher.fabric8.io/docs/vertx-runtime.html#creating-an-application_vertx

Reply:
Ah, of course. Thank you, @spisiakm, I'll file a follow-up PR to remove the versions.